README.md

```diff
@@ -1,210 +1,104 @@
-    Microbench.justByteArray     ALWAYS_SPEC       100       4  avgt    3    0.012 ±  0.011  us/op
-    Microbench.justByteArray   ALWAYS_UNSAFE       100       4  avgt    3    0.013 ±  0.004  us/op
-    Microbench.justByteArray           NEVER       100       4  avgt    3    0.055 ±  0.010  us/op
-    Microbench.justByteBuffer    ALWAYS_SPEC       100       4  avgt    3    0.942 ±  0.230  us/op
-    Microbench.justByteBuffer  ALWAYS_UNSAFE       100       4  avgt    3    0.935 ±  0.154  us/op
-    Microbench.justByteBuffer          NEVER       100       4  avgt    3    0.689 ±  0.247  us/op
-    Microbench.justString        ALWAYS_SPEC       100       4  avgt    3    0.616 ±  0.165  us/op
-    Microbench.justString      ALWAYS_UNSAFE       100       4  avgt    3    0.684 ±  0.636  us/op
-    Microbench.justString              NEVER       100       4  avgt    3    0.574 ±  0.098  us/op
-
-
-The fine print
-==============
-
-`MemoryMeter` can use
-`java.lang.instrument.Instrumentation.getObjectSize()`, which only claims
-to provide "approximate" results, but in practice seems to work as
-expected.
-
-Two more implementations are available as well. One is using
-`sun.misc.Unsafe` w/ reflection and one just uses reflection. Both are
-inaccurate and rely on guesses. It is strongly recommended to *not
-use* "unsafe" or "specification" and jamm will emit a warning on
-stderr:
-* The "Unsafe" implementation performs arithmetics on the "cookies"
-  returned by `Unsafe.objectFieldOffset()`, althought the Javadoc says:
-  `Do not expect to perform any sort of arithmetic on this offset;
-  it is just a cookie which is passed to the unsafe heap memory accessors.`
-  The implementation does not always consider Java object layouts in under
-  all circumstances for all JVMs.
-* The "specification" is just a best-effort guess-timate that can
-  easily produce results that are wrong w/ newer JVMs and GCs and/or
-  newer Java features.
-
-0.4.0 notes
-===========
-
-* The method `MemoryMeter.countChildren()` has been removed, because
-  there is a) no corresponding implementation from JDK-8249196 and
-  b) there was no test coverage at all.
-* The construction of the `MemoryMeter` instance has been refactored
-  to a proper builder pattern to allow distinct classes for each
-  implementation: Instrumentation, Unsafe, Specification).
-* A bunch of internal optimizations went into the existing guess-modes
-  and performance for these modes has also been _significantly_
-  improved (about 10x, see microbenchmark results above).
-* Support for `@Unmetered` for fields has been removed for the future
-  implementation of `java.lang.Runtime.deepSizeOf()` with Java 16.
+## Debugging
+
+In order to see the object tree visited when calling `MemoryMeter.measureDeep` and ensuring that it matches your
+expectations you can build the `MemoryMeter` instance using `printVisitedTree`:
+
+```
+    MemoryMeter meter = MemoryMeter.builder().printVisitedTree().build();
+```
```

Comment:
Do we want to add a note that benchmark tests exist and how they can be tested. As tests folder is visible in general, but I think it is good to stress that there is performance testing available. I also created initial test image and first CircleCI pipeline with all tests but jmh (I will add them  too on Tuesday). 
JDK17 tests are failing in the pipeline now as it turned there is some issue with mvn version that gets installed in Ubuntu with JDK17... I will fix it on Tuesday. This is just "preview", let me know what you think. I am off tomorrow, will continue on Tuesday 
In the future we can start adding builds also with 11, 17, not only 8 but this is not a thing we need at this moment. 
Posted the links in the weekly update but also here for your consideration 
- docker image - https://hub.docker.com/r/edimitrova86/jamm-testing/tags
- CircleCI config - https://github.com/ekaterinadimitrova2/jamm/tree/review-final/.circleci 
- pipeline - https://app.circleci.com/pipelines/github/ekaterinadimitrova2/jamm/6/workflows/f14b7263-669a-4546-8bb4-e28551efe0b6